### PR TITLE
Admin confirm

### PR DESCRIPTION
**Pull request by kotita49**
Admin page with a list of sightings to confirm

### CODE REVIEW

File: Models/Database/Sighting.cs [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
 using whale_spotting.Models.ApiModels;
 
 namespace whale_spotting.Models.Database
 {
+    public enum ConfirmState {
+        Review,
+        Confirmed,
+        Deleted
+    }
     public class Sighting
```
Comment (chloejonessoftwire) [inline Models/Database/Sighting.cs:15]: your bracket style changes here

File: ClientApp/src/Api/apiClient.ts [hunk 2]
```diff
@@ -9,20 +9,47 @@
...
+  });
 
   if (!response.ok) {
     throw new Error(await response.json());
   }
 }
+
+export async function fetchUnconfirmedSightings(): Promise<
+  ConfirmListResponse<Sighting>
+> {
```
Comment (chloejonessoftwire) [inline ClientApp/src/Api/apiClient.ts]: is the indentation okay here?

Comment (callumfg) [reply ClientApp/src/Api/apiClient.ts]: shift-alt-f processed it like this

File: Models/Database/Sighting.cs (hunk 1 above)
Comment (callumfg) [reply Models/Database/Sighting.cs:15]: sorted

File: ClientApp/src/Pages/ConfrimSighting/ConfirmSighting.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+export function ListOfUnconfirmed(): JSX.Element {
+  const [
+    unconfirmedSightingsData,
+    setUnconfirmedSightingsData,
+  ] = useState<null | ListSightings>(null);
+
+  useEffect(() => {
+    fetch(`api/confirm-sighting`)
```
Comment (ChristianMcc) [inline ClientApp/src/Pages/ConfrimSighting/ConfirmSighting.tsx]: Can we use the API client here instead please? might need some figuring out of how it works - emily and roshni have a PR open that uses it in a similar way :)

File: ClientApp/src/Api/apiClient.ts [hunk 4]
```diff
@@ -9,20 +9,47 @@
...
+  latitude: number;
+  longitude: number;
+  description: string;
+  sightedAt: string;
+  submittedByName: string;
+  submittedByEmail: string;
+  confirmState: number;
+}
+export interface ConfirmListResponse<T> {
+  items: T[];
```
Comment (ChristianMcc) [inline ClientApp/src/Api/apiClient.ts]: I don't think we need this? This is like a paginated list response, we just want a list of Sightings in our List response  and not a generic list with totalNumberOfItems/page/nextPage etc. (like we have below in ListSightings)

**Review by ChristianMcc — COMMENTED** [review]
Couple of comments about the apiclient side of things :)

File: ClientApp/src/Pages/ConfrimSighting/ConfirmSighting.tsx (hunk 3 above)
Comment (callumfg) [reply ClientApp/src/Pages/ConfrimSighting/ConfirmSighting.tsx]: Have changed this to use the function in api client

File: ClientApp/src/Api/apiClient.ts (hunk 4 above)
Comment (callumfg) [reply ClientApp/src/Api/apiClient.ts]: We have deleted this

File: Startup.cs [hunk 5]
```diff
@@ -66,7 +66,7 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
                 {
                     endpoints
                         .MapControllerRoute(name: "default",
-                        pattern: "{controller}/{action=Index}/{id?}");
+                        pattern: "api/{controller}/{action=Index}/{id?}");
```
Comment (ChristianMcc) [inline Startup.cs]: I don't think this ended up working when we overrode the Route on the controller that all the controllers are currently doing, so let's delete this change to not cause confusion :)

**Review by ChristianMcc — APPROVED** [review]
Left one comment to delete something - once that's resolved feel free to merge :)

**Pull request merged**